vignettes/syntax.Rmd

```diff
@@ -0,0 +1,398 @@
+---
+title: "The tidyselect syntax"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{syntax}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+options(
+  tibble.print_min = 4,
+  tibble.print_max = 4
+)
+options(
+  crayon.enabled = FALSE
+)
+```
+
+This is a technical description of the tidyselect syntax.
+
+```{r setup}
+library(tidyselect)
+library(magrittr)
+
+# FIXME: `select_pos()` should be exported as `eval_select()`
+select_pos <- function(data, ...) {
+  tidyselect:::select_pos(data, !!rlang::expr(c(...)))
+}
+
+# For better printing
+mtcars <- tibble::as_tibble(mtcars)
+iris <- tibble::as_tibble(iris)
+```
+
+
+## Sets of variables
+
+The tidyselect syntax is all about __sets__ of variables, internally
+represented by integer vectors of __positions__. `c(1L, 2L, 1L)`
```

Comment:
For example, `c(1L, 2L)` represents the first and second variables, as does `c(1L, 2L, 1L)`.